src/client/Client.cc

```diff
@@ -8916,12 +8923,26 @@ loff_t Client::_lseek(Fh *f, loff_t offset, int whence)
     break;
 
   case SEEK_END:
-    r = _getattr(in, CEPH_STAT_CAP_SIZE, f->actor_perms);
-    if (r < 0)
-      return r;
     pos = in->size + offset;
     break;
 
+  case SEEK_DATA:
+    if (offset >= in->size) {
```

Comment:
offset is a signed value, so I think you need to check for it being negative as well. The kclient's llseek implementation does this for SEEK_DATA and SEEK_HOLE:
```
                if (offset < 0 || offset >= i_size) {
                        ret = -ENXIO;
                        goto out;
                }
```
I think you'll want to do the same here.

Reply:
> offset is a signed value, so I think you need to check for it being negative as well. The kclient's llseek implementation does this for SEEK_DATA and SEEK_HOLE:
> 
> ```
>                 if (offset < 0 || offset >= i_size) {
>                         ret = -ENXIO;
>                         goto out;
>                 }
> ```
> 
> I think you'll want to do the same here.

Thanks. Changed